modules/search/instant-search/lib/query-string.js

```diff
@@ -16,11 +16,15 @@ import {
 import { getFilterKeys, getUnselectableFilterKeys, mapFilterToFilterKey } from './filters';
 import { decode } from '../external/query-string-decode';
 
-function getQuery() {
+export function getQuery() {
 	return decode( window.location.search.substring( 1 ), false, false );
 }
 
-function pushQueryString( queryString, shouldEmitEvent = true ) {
+export function setQuery( queryObject ) {
+	pushQueryString( encode( queryObject ) );
+}
+
+function pushQueryString( queryString ) {
```

Comment:
In hindsight, these changes should have landed in #18118. However, this isn't a big concern given that the `effects.js` module doesn't get loaded until it's integrated with the component tree.